cmd/root.go

```diff
@@ -122,17 +124,23 @@ the latest stable version will be downloaded from the repository.`,
 			originHelpFunc(cmd, args)
 			return
 		}
+		env, err := environment.InitEnv(repoOpts)
 		cmd, n, e := cmd.Root().Find(args)
-		if (cmd == rootCmd || e != nil) && len(n) > 0 {
+		if (cmd == rootCmd || e != nil) && len(n) > 0 && err != nil {
 			externalHelp(env, n[0], n[1:]...)
 		} else {
 			cmd.InitDefaultHelpFlag() // make possible 'help' flag to be shown
 			_ = cmd.Help()
 		}
+		if err == nil {
```

Comment:
Do we need to set the global env? I think there will not be used in the subsequent.